ggshield/cmd/auth/auth_config.py

```diff
@@ -0,0 +1,169 @@
+from typing import Optional
+
+import click
+
+from ggshield.core.config import Config, UnknownInstanceError
+
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+FIELD_OPTIONS = ("default_token_lifetime",)
+
+
+@click.command()
+@click.pass_context
+def config_list_cmd(ctx: click.Context) -> int:
+    """
+    List all auth configurations. Example of output
+
+    [default]
+    default_token_lifetime: 14
+    workspace: 23
+    url: *dashboard.gitguardian.com*
+    token: *62890f237c703c92fbda8236ec2a055ac21332a46115005c976d68b900535fb5*
+    token_name: *my_token*
+    token_type: *personal_access_token*
+    expiry: *2022-02-23T12:34:56Z*
+    """
+    config: Config = ctx.obj["config"]
+
+    message = ""
+    for instance in config.auth_config.instances:
+        instance_name = instance.name or instance.url
+
+        if instance.account is not None:
+            account_id = instance.account.account_id
+            token = instance.account.token
+            token_name = instance.account.token_name
+            expire_at = instance.account.expire_at
+            expiry = (
+                expire_at.strftime(DATETIME_FORMAT)
+                if expire_at is not None
+                else "never"
+            )
+        else:
+            account_id = token = token_name = expiry = "not set"  # type: ignore
+
+        message_lines = [
+            f"[{instance_name}]",
+            f"default_token_lifetime: {instance.default_token_lifetime}",
+            f"workspace: {account_id}",
+            f"url: {instance.url}",
+            f"token: {token}",
+            f"token_name: {token_name}",
+            f"expiry: {expiry}",
+        ]
+        message += ("\n".join(message_lines)) + "\n\n"
+
+    click.echo(message[:-2])
+    return 0
+
+
+@click.command()
+@click.argument("field_name", nargs=1, type=click.Choice(FIELD_OPTIONS), required=True)
+@click.argument("value", nargs=1, type=click.STRING, required=True)
+@click.option(
+    "--instance",
+    required=False,
+    type=str,
+    help="URL of the instance to set the config.",
+)
+@click.pass_context
+def config_set_command(
+    ctx: click.Context, field_name: str, value: str, instance: Optional[str]
+) -> int:
```

Comment:
Since we are exposing `account_id` as `workspace`, we must translate it here. But I think we should actually rename the field from `account_id` to `workspace` in the config file, would save us some trouble.